### ISSUE TITLE
Allow editing of a user's own profile

### ISSUE THREAD

**Issue by timmyichen** [timmyichen/dev-directory#12]
Project: #19 
Depends on: #10 

**Description**
If the user is logged in and viewing their own profile, they should be able to edit it.  Add an edit button to the page that, when clicked, makes the page into a form where they can edit their editable information.  Use the PATCH route created in #10 when submitting.  Only editable fields should appear in the form.  Once successfully saved, it should leave "edit" mode and show the newly updated information.

Out of scope: Error handling / surfacing errors on the client.

**Acceptance Criteria**
* Edit button exists only when a user's viewing their own profile
* Loading state exists once the PATCH request is sent (prevent more than one inflight request)
* New information is displayed once saved.
* e2e testing
